packages/core/src/components/search/SearchInput/SearchInput.tsx

```
@@ -77,28 +143,80 @@ const SearchInput = forwardRef<SearchInputRef, SearchInputProps>(
+    const csvParser = useCSVParser(csvParserOptions)
+    const {
+      error: csvError,
+      isParsing: isCsvProcessing,
+      onParseFile,
+      onClearError,
+      onGenerateTemplate,
+    } = csvParser
+    // Access globalSettings for fileUpload configuration (section and content-type)
+    let fileUploadConfig
+    try {
+      const pageContext = usePage<{ globalSettings?: { fileUpload?: any } }>()
```

Biome (2.4.4)

[error] This hook is being called conditionally, but all hooks must be called in the exact same order in every component render.

(lint/correctness/useHookAtTopLevel)